server/src/utils/general-controller.ts

```diff
@@ -10,30 +10,53 @@ class GeneralController {
         this.name = name;
     }
 
+    /**
+     * Creates new item in model
+     * @param params {object}
+     * @param params.data {object} data to store
+     * @returns {Promise<object>} created item
+     */
     async create(params: any) {
-        const newEvent = new this.model(params.data);
-        if (await this.model.create(newEvent)) {
-            console.log(allowedFields[this.name]);
-            const event = await this.model.findById(newEvent._id, allowedFields[this.name]);
-            return event;
+        const newItem = new this.model(params.data);
+        if (await this.model.create(newItem)) {
+            const item = await this.model.findById(newItem._id, allowedFields[this.name]);
+            return item;
         }
-        throw new Error('Problem with creating event');
+        throw new Error(`Problem with creating item in ${this.name}`);
 
     }
 
+    /**
+     * Returns item from model based on id
+     * @param params {object}
+     * @param params.id {string} id of item to find
+     * @returns {Promise<Object>}
+     */
     async get(params: any) {
-        const event = await this.model.findById(params.id, allowedFields[this.name]);
-        if (event) {
-            return event;
+        const item = await this.model.findById(params.id, allowedFields[this.name]);
```

Comment:
As far as I know the projection can be omitted if we want to return all fields (e.g. the specified allowed fields for User contains all User's fields), or it is used differently?

Reply:
currently mongo returns only fields that are specified in allowed fields... I think if we want to change that, it should be done in separate ticket